src/crimson/osd/osd.cc

```diff
@@ -608,6 +610,41 @@ seastar::future<> OSD::_send_boot()
   return monc->send_message(std::move(m));
 }
 
+seastar::future<> OSD::_add_device_class()
+{
+  LOG_PREFIX(OSD::_add_device_class);
+  if (!local_conf().get_val<bool>("osd_crush_update_on_start")) {
+    return seastar::now();
+  }
+
+  auto get_device_class = [this]() -> seastar::future<std::string> {
+    return store.get_default_device_class().then([FNAME, this](std::string device_class) {
+      return seastar::make_ready_future<std::string>(device_class);
+    }).handle_exception([FNAME](std::exception_ptr e) {
+      ERROR("Failed to get device class: ", e);
+      return seastar::make_ready_future<std::string>("");
+    });
+  };
+
+  return get_device_class().then([FNAME, this](auto device_class) {
```

Comment:
We should check that `device_class` is not an empty string before using it

Reply:
done